voorhoede/head-start
# feature: API routes

# Changes

- Adds support for dynamic API routes on Cloudflare (using `@astro/cloudflare` to enable Cloudflare Workers).
- Adds access to Cloudflare runtime object in pages (`Astro.locals.runtime`) and API routes (`context.locals.runtime`).
- Extends local preview setup with Cloudflare runtime support.
- Upgrades from Astro v2 to Astro v3 since it's stable and introduces a new API route syntax.
- Adds demo endpoints
- Redirects root url to home page in user's preferred language. Uses a dynamic endpoint in the page root and parses the language accept header.

# Associated issue

- Resolves #28 
- Enables / part of #23 

# How to test

1. Open preview link
2. Verify you are redirected to home page in your preferred (browser/system) locale (or to the default locale if yours isn't available).
3. Test API endpoint using URL `searchParams`: `/api/demo/hello?to=you`
4. Verify it responds with dynamic JSON (you can change the `to` value)
5. Test API endpoint using `request.cf` props: `/api/demo/geo`
6. Verify it responds with dynamic JSON (your location)

# Checklist

- ~~I have added/updated tests to prove that my feature works (if not possible please explain why)~~
- ~~I have added a decision log entry if the change affects the architecture or changes a significant technology~~
